run_network_test.go

```diff
@@ -146,46 +146,120 @@ func TestRunPort(t *testing.T) {
+		connectURLPort int
+		err            string
 	}
 	lo := net.ParseIP("127.0.0.1")
 	zeroIP := net.ParseIP("0.0.0.0")
 	testCases := []testCase{
 		{
-			listenIP:  lo,
-			connectIP: lo,
-			port:      8080,
+			listenIP:       lo,
+			connectIP:      lo,
+			hostPort:       "8080",
+			containerPort:  "80",
+			connectURLPort: 8080,
 		},
 		{
 			// for https://github.com/containerd/nerdctl/issues/88
-			listenIP:  hostIP,
-			connectIP: hostIP,
-			port:      8080,
+			listenIP:       hostIP,
+			connectIP:      hostIP,
+			hostPort:       "8080",
+			containerPort:  "80",
+			connectURLPort: 8080,
 		},
 		{
-			listenIP:  hostIP,
-			connectIP: lo,
-			port:      8080,
-			err:       "connection refused",
+			listenIP:       hostIP,
+			connectIP:      lo,
+			hostPort:       "8080",
+			containerPort:  "80",
+			connectURLPort: 8080,
+			err:            "connection refused",
 		},
 		{
-			listenIP:  lo,
-			connectIP: hostIP,
-			port:      8080,
-			err:       "connection refused",
+			listenIP:       lo,
+			connectIP:      hostIP,
+			hostPort:       "8080",
+			containerPort:  "80",
+			connectURLPort: 8080,
+			err:            "connection refused",
 		},
 		{
-			listenIP:  zeroIP,
-			connectIP: lo,
-			port:      8080,
+			listenIP:       zeroIP,
+			connectIP:      lo,
+			hostPort:       "8080",
+			containerPort:  "80",
+			connectURLPort: 8080,
 		},
 		{
-			listenIP:  zeroIP,
-			connectIP: hostIP,
-			port:      8080,
+			listenIP:       zeroIP,
+			connectIP:      hostIP,
+			hostPort:       "8080",
+			containerPort:  "80",
+			connectURLPort: 8080,
+		},
+		{
+			listenIP:       lo,
+			connectIP:      lo,
+			hostPort:       "7000-7005",
+			containerPort:  "79-84",
+			connectURLPort: 7001,
+		},
+		{
+			listenIP:       hostIP,
+			connectIP:      hostIP,
+			hostPort:       "7000-7005",
+			containerPort:  "79-84",
+			connectURLPort: 7001,
+		},
+		{
+			listenIP:       hostIP,
+			connectIP:      lo,
+			hostPort:       "7000-7005",
+			containerPort:  "79-84",
+			connectURLPort: 7001,
+			err:            "connection refused",
+		},
+		{
+			listenIP:       lo,
+			connectIP:      hostIP,
+			hostPort:       "7000-7005",
+			containerPort:  "79-84",
+			connectURLPort: 7001,
+			err:            "connection refused",
+		},
+		{
+			listenIP:       zeroIP,
+			connectIP:      hostIP,
+			hostPort:       "7000-7005",
+			containerPort:  "79-84",
+			connectURLPort: 7001,
+		},
+		{
+			listenIP:       zeroIP,
+			connectIP:      lo,
+			hostPort:       "7000-7005",
+			containerPort:  "80-85",
+			connectURLPort: 7001,
+			err:            "error after 30 attempts",
+		},
+		{
+			listenIP:       zeroIP,
+			connectIP:      lo,
+			hostPort:       "7000-7005",
+			containerPort:  "80",
+			connectURLPort: 7000,
+		},
+		{
+			listenIP:       zeroIP,
+			connectIP:      lo,
+			hostPort:       "7000-7005",
+			containerPort:  "80",
+			connectURLPort: 7005,
+			err:            "connection refused",
 		},
```

Comment:
Can we also test an invalid range such as `-p 7000-7005:79-85`

Reply:
done